docs/DevGuide/Parallelization.md

```diff
@@ -0,0 +1,539 @@
+needed to initialize the parallel component. These options are specified in the
+`add_options_to_databox` type alias of the parallel component. The `initialize`
+functions of different parallel components are called in random order and so it
+is not safe to have them depend on each other.
+
+Each parallel component must also decide what to do in the different phases of
+the execution. This is controlled by an `execute_next_phase` function with
+signature:
+\code
+static void execute_next_phase(
+    const typename metavariables::Phase next_phase,
+    const Parallel::CProxy_ConstGlobalCache<metavariables>& global_cache);
+\endcode
+The `determine_next_phase` function in the Metavariables determines the next
+phase, after which the `execute_next_phase` function gets called. The
+`execute_next_phase` function determines what the parallel component should do
+during the next phase. Typically the `execute_next_phase` function should just
+call `start_next_phase(phase)` on the parallel component. In the future
+`execute_next_phase` may be removed.
+
+An example of a singleton parallel component is:
+\snippet Test_AlgorithmParallel.cpp singleton_parallel_component
+
+An example of an array parallel component is:
+\snippet Test_AlgorithmParallel.cpp array_parallel_component
+Elements are inserted into the array by using the Charm++ `insert` member
+function of the CProxy for the array. The `insert` function is documented in
+the Charm++ manual. In the above Array example `array_proxy` is a `CProxy` and
+so all the documentation for Charm++ array proxies applies. SpECTRE always
+creates empty arrays with the constructor and requires users to insert however
+many elements they want and on which cores they want them to be placed. Note
+that load balancing calls may result in array elements being moved.
+
+# Actions {#dev_guide_parallelization_actions}
+
+For those familiar with Charm++, actions should be thought of as effectively
+being entry methods. They are functions that can be invoked on a remote object
+(chare/parallel component) using a `CProxy` (see the [Charm++
+manual](http://charm.cs.illinois.edu/help)), which is retrieved from the
+ConstGlobalCache using the parallel component struct and the
+`Parallel::get_parallel_component()` function. %Actions are structs with a
+static `apply` method and come in three variants: simple actions, iterable
+actions, and reduction actions. One important thing to note
+is that actions cannot return any data to the caller of the remote method.
+Instead, "returning" data must be done via callbacks or a callback-like
+mechanism.
+
+The simplest signature of an `apply` method is for iterable actions:
+\snippet Test_AlgorithmCore.cpp apply_iterative
+The return type is discussed at the end of each section describing a particular
+type of action. Simple actions can have additional arguments but must have at
+least the arguments shown above. Reduction actions must have the above arguments
```

Comment:
I believe this is no longer true as they no longer have access to the Inbox or ActionList?